src/array/cuda/bf16.cuh

```diff
@@ -0,0 +1,121 @@
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 800
+  return __hmax(a, b);
+#else
+  return __nv_bfloat16(max(float(a), float(b)));  // NOLINT
+#endif
+}
+
+static __device__ __forceinline__ __nv_bfloat16 min(__nv_bfloat16 a, __nv_bfloat16 b) {
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 800
+  return __hmin(a, b);
+#else
+  return __nv_bfloat16(min(float(a), float(b)));  // NOLINT
+#endif
+}
+
+#ifdef __CUDACC__
+// Arithmetic BF16 operations for architecture >= 8.0 are already defined in cuda_bf16.h
+#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 800)
+__device__ __forceinline__ __nv_bfloat16 operator+(
+    const __nv_bfloat16& lh, const __nv_bfloat16& rh) {
+  return __nv_bfloat16(float(lh) + float(rh));  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16 operator-(
+    const __nv_bfloat16& lh, const __nv_bfloat16& rh) {
+  return __nv_bfloat16(float(lh) - float(rh));  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16 operator*(
+    const __nv_bfloat16& lh, const __nv_bfloat16& rh) {
+  return __nv_bfloat16(float(lh) * float(rh));  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16 operator/(
+    const __nv_bfloat16& lh, const __nv_bfloat16& rh) {
+  return __nv_bfloat16(float(lh) / float(rh));  // NOLINT
+}
+
+__device__ __forceinline__ __nv_bfloat16& operator+=(
+    __nv_bfloat16& lh, const __nv_bfloat16& rh) {  // NOLINT
+  lh = __nv_bfloat16(float(lh) + float(rh)); return lh;  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16& operator-=(
+    __nv_bfloat16& lh, const __nv_bfloat16& rh) {  // NOLINT
+  lh = __nv_bfloat16(float(lh) - float(rh)); return lh;  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16& operator*=(
+    __nv_bfloat16& lh, const __nv_bfloat16& rh) {  // NOLINT
+  lh = __nv_bfloat16(float(lh) * float(rh)); return lh;  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16& operator/=(
+    __nv_bfloat16& lh, const __nv_bfloat16& rh) {  // NOLINT
+  lh = __nv_bfloat16(float(lh) / float(rh)); return lh;  // NOLINT
+}
+
+__device__ __forceinline__ __nv_bfloat16& operator++(__nv_bfloat16& h) {  // NOLINT
+  h = __nv_bfloat16(float(h) + 1.0f); return h;  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16& operator--(__nv_bfloat16& h) {  // NOLINT
+  h = __nv_bfloat16(float(h) - 1.0f); return h;  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16  operator++(__nv_bfloat16& h, int) {  // NOLINT
+  __nv_bfloat16 ret = h; h = __nv_bfloat16(float(h) + 1.0f); return ret;  // NOLINT
+}
+__device__ __forceinline__ __nv_bfloat16  operator--(__nv_bfloat16& h, int) {  // NOLINT
+  __nv_bfloat16 ret = h; h = __nv_bfloat16(float(h) - 1.0f); return ret;  // NOLINT
+}
+
+__device__ __forceinline__ __nv_bfloat16 operator+(const __nv_bfloat16& h) { return h; }
+__device__ __forceinline__ __nv_bfloat16 operator-(const __nv_bfloat16& h) {
+  return __nv_bfloat16(-float(h));  // NOLINT
+}
+
+__device__ __forceinline__ bool operator==(const __nv_bfloat16& lh, const __nv_bfloat16& rh) {
```

Comment:
I think we are following Google C++ code style. 
Which advocate 80 characters line.
https://google.github.io/styleguide/cppguide.html#Line_Length